src/JarvisDesk.cpp

```diff
@@ -107,26 +104,46 @@ class JarvisDesk_impl {
     }
     else pending_stop = false;
 
-    if (latch_timer.trigger()) {
+    if (latch_timer.trigger())
       unlatch();
-    }
 
     // Publish pending updates periodically to AdafruitIO
     io_send();
   }
 
+  void press_Memory(int duration = 30) {
+#if not defined(JCB35N2PA32V2) // JCB35N2PA32V2 doesn't use this line when pressing memory
+    latch_pin(HS0);
+#endif
+    latch_pin(HS3);
+    latch_timer.reset(duration);
+    Log.println("Pressing memory for ", duration, "ms");
+  }
+
   bool goto_preset(int p) {
-    if (!p || p >4) return false;
-    if (p == preset) return false;
+    if (!p || p > 4)
+      return false;
+    if (p == preset) {
+      Log.println("Already at preset");
+      return false;
+    }
     pending_preset = p;
+    Log.println("Goto preset ", p);
     return true;
   }
 
+  void reset() {
+    // Press and hold down for 10 seconds
+    latch_pin(HS0);
+    latch_timer.reset(5000);
```

Comment:
5 seconds?  Or 10?

Reply:
Moving to reset(int duration = 5000), this is the length of time that it takes to clear most errors. but may not be enough to re-level the desk

Reply:
Resolved in my last commit.